### PR TITLE
Add proper local tx submission to the demo

### PR DESCRIPTION
**Pull request by dcoutts**
Remove the mock impl of the local tx submission and add add the proper local tx submission client.

The CLI changes slightly, we have to pass the protocol of the system we're using (since we now do the mock tx to real tx elaboration on the client side rather than the server side).

### CODE REVIEW

File: ouroboros-consensus/demo-playground/TxSubmission.hs [hunk 1]
```diff
@@ -73,8 +104,126 @@ command' c descr p =
...
+                   -> TopologyInfo
+                   -> Mock.Tx
+                   -> IO ()
+handleTxSubmission p tinfo mocktx = do
+    topoE <- readTopologyFile (topologyFile tinfo)
+    case topoE of
+         Left e  -> error e
+         Right t@(NetworkTopology nodeSetups) -> do
+
+             case M.lookup (node tinfo) (toNetworkMap t) of
```
Comment (mrBliss) [inline ouroboros-consensus/demo-playground/TxSubmission.hs]: Maybe `member` or `notMember` would make the intent clearer: just checking presence, we don't need the actual value associated with the key.

File: ouroboros-consensus/demo-playground/TxSubmission.hs [hunk 2]
```diff
@@ -73,8 +104,126 @@ command' c descr p =
...
+                  Nothing -> fail "Target node not found."
+                  Just _  -> return ()
+
+             let CoreId nid = node tinfo
```
Comment (mrBliss) [inline ouroboros-consensus/demo-playground/TxSubmission.hs]: I'd rather have a pattern-match + error to avoid the warning.

Comment (dcoutts) [reply ouroboros-consensus/demo-playground/TxSubmission.hs]: done

File: ouroboros-consensus/demo-playground/TxSubmission.hs (hunk 1 above)
Comment (dcoutts) [reply ouroboros-consensus/demo-playground/TxSubmission.hs]: agree

**Pull request merged**